doc/reference/classes/index.rst

```diff
@@ -73,6 +73,103 @@ Core Views
    FilterMultiInner
    FilterMultiAdjacency
 
+Reporting Views
+===============
+
+NetworkX provides lightweight *reporting views* — read-only objects that
+report information about a graph without copying it. The most commonly used
+reporting views are:
+
+- ``NodeView`` — returned by ``G.nodes`` or ``G.nodes(...)``.
+- ``EdgeView`` — returned by ``G.edges`` or ``G.edges(...)``.
+- ``DegreeView`` — returned by ``G.degree`` or ``G.degree(...)``.
+
+Quick overview
+--------------
+Views are quick-to-create, live (they reflect changes to the graph), and
+provide Pythonic access patterns:
+
+- set-like membership and set operations for nodes and edges (``n in G.nodes``,
+  ``G.nodes & H.nodes``, ``(u, v) in G.edges``),
+- iteration (``for n in G.nodes``, ``for u, v in G.edges``),
+- mapping-style lookups (``G.nodes[n]`` returns the node attribute dict;
+  ``G.edges[u, v]`` returns the edge attribute dict),
+- data-filtered iteration via ``.data(...)`` and conversion to concrete
+  containers via ``list(...)`` or ``dict(...)``.
+
+Common usage patterns
+---------------------
+**NodeView / NodeDataView**
+
+- Use ``G.nodes`` when you need membership tests, set-like operations, or to
+  look up a node's attribute dict with ``G.nodes[n]``.
+- Use ``G.nodes(data=...)`` or ``G.nodes.data(key, default=...)`` to iterate
+  node/data pairs or to extract a single attribute for all nodes.
+
+Example:
+
+.. code-block:: python
+
+    >>> G = nx.path_graph(3)
+    >>> list(G.nodes)
+    [0, 1, 2]
+    >>> G.add_node(3, color="red")
+    >>> list(G.nodes.data("color", default=None))
+    [(0, None), (1, None), (2, None), (3, 'red')]
+
+**EdgeView / EdgeDataView**
+
+- Use ``G.edges`` to iterate or test membership for edges.
+- Call ``G.edges(data=...)`` or ``G.edges(nbunch=..., data=..., keys=...)`` to
+  iterate edges with data, restrict to edges incident to a set of nodes, or
+  include multigraph keys.
+
```

Comment:
Could add something like:
```
- Note: Iteration of `G.edges()` yields node pairs as 2-tuples even for multigraphs, while `G.edges` (without the function form) yields 3-tuples `(u, v, edge_key)` for multigraphs and 2-tuples for non-multigraphs. This can allow some code to work with edges for either graph type when the `edge_key` is not critical.
```

But perhaps this is too much detail -- still I'd like to see it somewhere. I get asked about it a lot. (sign of an API wart I suspect.)